packages/api/posthog-client.mts

```diff
@@ -0,0 +1,80 @@
+import { PostHog } from 'posthog-node';
+import { getConfig } from './config.mjs';
+import { IS_PRODUCTION } from './constants.mjs';
+
+type QueuedEvent = {
+  event: string;
+  properties?: Record<string, any>;
+};
+
+class PostHogClient {
+  private static instance: PostHogClient | null = null;
+  private distinctId: string | null = null;
+  private client: PostHog | null = null;
+  private isEnabled: boolean = false;
+  private isInitialized: boolean = false;
+  private initializationPromise: Promise<void> | null = null;
+  private eventQueue: QueuedEvent[] = [];
+
+  private constructor() {
+    this.initializationPromise = this.initialize();
+  }
+
+  public static getInstance(): PostHogClient {
```

Comment:
I think we can remove a lot of this code if we just use top-level await to get the config, define a class that takes the config in the constructor, and export an instance of said class?

Reply:
done. I initially wanted to avoid that because top level await bit us with initialization before.. but since you had the same inclination, happy to go that direction